### PR TITLE
✨ Add intervention disabling functionality

### LINKED ISSUE #62: Add intervention disabling option

### PR DESCRIPTION
**Pull request by uoa-ypec413**
## Description

Adds the ability to disable interventions when on the third question
-> A popup appears when question 3 is opened for the first time
-> Interventions can then be toggled by clicking the settings button in the app header

Closes #62

Note(s):
- The modal is hard coded to pop up on question 3. I won't be adding functionality to bring it up on later questions as I've made the assumption that we won't be having over 3 questions. Let me know if you're concerned about this.

## How has this been tested?

Locally 😅 

## Screenshots

image
image
image

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass

### CODE REVIEW

File: src/hooks/useInterventions.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  useEffect(() => {
+    if (userDocumentData !== undefined) {
+      if (userDocumentData.interventions !== undefined) {
+        setInterventions(userDocumentData.interventions);
+      } else if (userDoc) {
+        updateDoc(userDoc, { interventions: defaultInterventions });
+      }
+    }
+  }, [userDocumentData]);
```
Comment (kand198) [inline src/hooks/useInterventions.tsx:47]: Interventions need to be related to a specific user, and a specific assignment, and a specific question.

The way it currently works, if I went to a different assignment/question then the intervention restrictions would stick.

There's currently a glitch (fixed by #64) where you can accidentally go back a question, so right now I'm stuck on Q2 with no way of changing my interventions to turn more back on.

**Review by kand198 — CHANGES_REQUESTED** [review]
Great work, it looks great and generally works really well.

Aside from my minor critique below, would we be able to add a submit button to the "implementing a solution" page, or something similar? We need this in case they disable the testing page.

Also, how easy would it be to re-direct the user rather than leave them on a 404 page if they deselect the "understanding the problem" page?

**Comment by uoa-ypec413** [comment]
> Also, how easy would it be to re-direct the user rather than leave them on a 404 page if they deselect the "understanding the problem" page?

Not hard but not easy. I think it's also important to give user feedback and that's what the 404 page does. I feel like it may get confusing if they deselect "understanding the problem" and then are redirected right away. What do you think @kand198 ?

File: src/hooks/useInterventions.tsx (hunk 1 above)
Comment (uoa-ypec413) [reply src/hooks/useInterventions.tsx:47]: The glitch will be fixed, and they should have no way of going back a question once that's fixed. So I don't see the point in having interventions related to a specific question. I agree that it should be related to a specific assignment though.

**Review by kand198 — APPROVED** [review]
Awesome work, well done!

**Pull request merged**